crates/rpc/src/handlers/config.rs

```diff
@@ -49,8 +32,8 @@ pub struct SpecConfig {
     inactivity_penalty_quotient: u64,
 }
 
-impl From<Arc<NetworkSpec>> for SpecConfig {
-    fn from(network_spec: Arc<NetworkSpec>) -> Self {
+impl From<&Arc<NetworkSpec>> for SpecConfig {
+    fn from(network_spec: &Arc<NetworkSpec>) -> Self {
```

Comment:
this is an arc we shouldn't need to reference it, it is already a pointer to shared memory

Reply:
Hi @SuccinctPaul I already tried doing this all locally, please implement my other suggestions first before coming to conclusions. I am sometimes wrong, and should be called out on it, but I did check locally that this works.